agent/src/main.rs

```diff
@@ -400,15 +404,17 @@ where
      */
     let expected_downstairs_instances = regions
         .iter()
-        .filter(|r| r.state == State::Created)
+        .filter(|r| r.state == State::Created || r.state == State::Requested)
         .map(|r| format!("{}-{}", downstairs_prefix, r.id.0))
         .collect::<HashSet<_>>();
 
     let expected_snapshot_instances: Vec<String> = running_snapshots
         .iter()
         .flat_map(|(_, n)| {
             n.iter()
-                .filter(|(_, rs)| rs.state == State::Created)
+                .filter(|(_, rs)| {
+                    rs.state == State::Created || rs.state == State::Requested
```

Comment:
Similarly, I don't think this is right, as it would start a downstairs process for a snapshot's `.zfs/snapshot/...` directory when that might not exist yet